### ISSUE TITLE
felicity.example() throws an error for Joi.object().default(<OBJECT>)

### ISSUE THREAD

**Issue by gxapplications** [xogroup/felicity#137]
#### Context

* *node version*: 7.10.1
* *felicity version*: 2.4.4
* *environment* (node, browser): nodejs / Linux
* *any relevant modules used with* (hapi, hoek, etc.): tests through mocha & chai/expect
* *any other relevant information*:

#### What are you trying to achieve or the steps to reproduce ?

When trying to generate an example with this, it's ok:
```
const schema = Joi.object().keys({
  a: Joi.object().default('hello string')
})
Felicity.example(schema)
```
-> OK

But when default() contains an object, it fails:
```
const schema = Joi.object().keys({
  a: Joi.object().default({ b: 'hello in an object' })
})
Felicity.example(schema)
```
-> KO (stack trace below)

#### What result did you expect ?

An example object with a clone of the object given in the default() function arg.

#### What result did you observe ?

```
TypeError: schemaDescription.flags.default.function is not a function
      at Object.getDefault (node_modules/felicity/lib/helpers.js:8:56)
      at ObjectExample._getDefaults (node_modules/felicity/lib/valueGenerator.js:87:24)
      at ObjectExample.generate (node_modules/felicity/lib/valueGenerator.js:55:25)
      at Object.keys.forEach (node_modules/felicity/lib/valueGenerator.js:653:48)
      at Array.forEach (native)
      at ObjectExample._generate (node_modules/felicity/lib/valueGenerator.js:628:53)
      at ObjectExample.generate (node_modules/felicity/lib/valueGenerator.js:63:21)
      at valueGenerator (node_modules/felicity/lib/valueGenerator.js:837:20)
      at Object.exampleGenerator [as example] (node_modules/felicity/lib/exampleGenerator.js:9:27)
```

**Comment by WesTyler** [comment]
Yep, sure enough! Thanks for the bug report. I'll see about getting it patched ASAP.

**Comment by WesTyler** [comment]
Published in v2.4.5 . Thanks again for checking out Felicity and for the report! I appreciate it! :D

**Comment by gxapplications** [comment]
Hey great, it works :)
Thank you for Felicity and your reactivity, I will digg a lot more in it because I will have a lot more complicated cases that will come soon... :)
